### ISSUE TITLE
save crashes with large wallet

### ISSUE THREAD

**Issue by fentanyluser** [turtlecoin/turtlecoin#1119]
wallet-api crashes when running save

**Comment by zpalmtree** [comment]
Is it repeatable? Can you provide a stack trace?

**Comment by fentanyluser** [comment]
yes it is repeatable, simply create a wallet with hundreds of thousands or millions of txes and trying running save

**Comment by brandonlehmann** [comment]
That's not good. Do you have copy of the wallet that successfully saved at *some* previous height? If so, it **may** be worth a try using https://github.com/turtlecoin/node-wallet-repair to force prune old transactions, spent inputs, etc. out of that wallet file (create a backup although the tool will do it for you). Once the wallet file is force pruned, it would start scanning from the last detected block height which may then allow you to save the wallet once it gets back to the top.

In any event, make sure you've got a backup of those keys/seed and a backup of the wallet file.

Out of curiosity though, how much memory do you have in the machine you're running this on?

**Comment by fentanyluser** [comment]
there is no issue with sending except that causes crash sometimes too, machine has 64gb of memory

**Comment by brandonlehmann** [comment]
> this will do nothing, there is no issue with sending except that causes crash sometimes too, machine has 64gb of memory. old turtle-service worked fine

I may be reading the stack trace wrong but it appears that it's failing to push memory into the vector memory heap. Considering that this *may* be the case, the easiest way to reduce the memory footprint and thus reduce the heap usage of the vector is to reduce the number of inputs stored in the wallet file (by pruning spent inputs and pruning data about transactions that are no longer relevant). The tool I linked you does just that (although it was created to assist with a different type of error) will help you with that piece so, I still advise to *give it a try*. 

You may also try, if you've kept the wallet *optimized* and have a reasonable belief that the funds in the wallet are further up in the chain, avoiding scanning from `0`.

If you're unwilling to give it a try, then you're kind of leaving it in a rough spot as *reproducing* an issue that is contingent on having *millions of transactions* (there are a little less than 8M total transactions in the chain today) is going to be *very* difficult to reproduce in any timely manner.

Note: turtle-service is no longer supported, no longer available, and it doesn't work with the nodes since Oct 2020. We're 6 months past that point.

**Comment by fentanyluser** [comment]
got about 100k blocks in with your tool after about 8 hours before it crashed again

**Comment by brandonlehmann** [comment]
Did the tool crash or did wallet-api crash? If it was the tool that crashed, did it give an error as to why?

As a quick temperature check, have you attempted to restore the wallet from keys using zedwallet directly? If so, optimize the wallet ASAP and send_all to yourself, then reset to a more recent scan height. It’ll cost a few TRTL to do so, but it’ll alleviate the issue. 

If that doesn’t work, there are few different ways to try to skin this cat to get the funds moving, optimizing the wallet, and getting it back to working order but it’s gonna take some work and possibly additional tools to do so. 

Rest assured, any funds are safe as long as you keep copies of the private spend and private view keys safe.

**Comment by fentanyluser** [comment]
have to open wallet and let is scan a bit, periodically try save until it crashes then load up your tool which took over 24hrs to go through <40k blocks at this rate it will be months

**Comment by brandonlehmann** [comment]
What is the output of the following on your system?

```
ulimit -v
ulimit -d
```

**Comment by fentanyluser** [comment]
both are unlimited

**Comment by fentanyluser** [comment]
zedwallet crashes also when scanning it tries to save progress

**Comment by fentanyluser** [comment]
```
#0  0x00005555556a5eef in std::vector<CryptoNote::KeyInput, std::allocator<CryptoNote::KeyInput> >::~vector (this=0xc1, __in_chrg=<optimized out>)
    at /usr/include/c++/8/bits/stl_vector.h:565
#1  0x00005555556a601a in WalletTypes::RawTransaction::~RawTransaction (this=0x41, __in_chrg=<optimized out>)
    at /turtlecoin/turtlecoin-1.0.0/include/WalletTypes.h:129
#2  0x00005555556a67ae in std::_Destroy<WalletTypes::RawTransaction> (__pointer=<optimized out>) at /usr/include/c++/8/bits/stl_construct.h:97
#3  0x00005555556a67d2 in std::_Destroy_aux<false>::__destroy<WalletTypes::RawTransaction*> (__first=0x41, __last=0x7fffc80c1ee0)
    at /usr/include/c++/8/bits/stl_construct.h:107
#4  0x00005555556a67e9 in std::_Destroy<WalletTypes::RawTransaction*> (__first=<optimized out>, __last=<optimized out>)
    at /usr/include/c++/8/bits/stl_construct.h:127
#5  0x00005555556a67f4 in std::_Destroy<WalletTypes::RawTransaction*, WalletTypes::RawTransaction> (__first=<optimized out>, __last=<optimized out>)
    at /usr/include/c++/8/bits/stl_construct.h:203
#6  0x00005555556a6816 in std::vector<WalletTypes::RawTransaction, std::allocator<WalletTypes::RawTransaction> >::~vector (this=0x7fffc81b7498,
    __in_chrg=<optimized out>) at /usr/include/c++/8/bits/stl_vector.h:565
#7  0x00005555556a683b in WalletTypes::WalletBlockInfo::~WalletBlockInfo (this=0x7fffc81b7430, __in_chrg=<optimized out>)
    at /turtlecoin/turtlecoin-1.0.0/include/WalletTypes.h:218
#8  0x00005555556a6853 in std::_Head_base<0ul, WalletTypes::WalletBlockInfo, false>::~_Head_base (this=this@entry=0x7fffc81b7430,
    __in_chrg=<optimized out>) at /usr/include/c++/8/tuple:120
#9  0x00005555556a6db5 in std::_Tuple_impl<0ul, WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int>::~_Tuple_impl (this=0x7fffc81b7410,
    __in_chrg=<optimized out>) at /usr/include/c++/8/tuple:185
#10 0x00005555556a6dcd in std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int>::~tuple (this=<optimized out>, __in_chrg=<optimized out>)
    at /usr/include/c++/8/tuple:556
#11 0x00005555556a6f32 in __gnu_cxx::new_allocator<std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int> >::destroy<std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int> > (this=this@entry=0x555577bf4d60, __p=<optimized out>) at /usr/include/c++/8/ext/new_allocator.h:140
#12 0x00005555556a6f3d in std::allocator_traits<std::allocator<std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int> > >::destroy<std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int> > (__a=..., __p=<optimized out>) at /usr/include/c++/8/bits/alloc_traits.h:486
#13 0x00005555556a6f58 in std::vector<std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int>, std::allocator<std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int> > >::pop_back (this=this@entry=0x555577bf4d60) at /usr/include/c++/8/bits/stl_vector.h:1112
#14 0x00005555556afc29 in std::priority_queue<std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int>, std::vector<std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int>, std::allocator<std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int> > >, OrderByArrivalIndex>::pop (60) at /usr/include/c++/8/bits/stl_queue.h:646
#15 0x00005555556afc37 in ThreadSafePriorityQueue<std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int>, OrderByArrivalIndex>::pop_unsafe (
    this=this@entry=0x555577bf4d60) at /turtlecoin/turtlecoin-1.0.0/src/utilities/ThreadSafePriorityQueue.h:115
#16 0x00005555556a1212 in WalletSynchronizer::mainLoop (this=0x555577bf4930)
    at /turtlecoin/turtlecoin-1.0.0/src/walletbackend/WalletSynchronizer.cpp:179
#17 0x00005555556a3e93 in std::__invoke_impl<void, void (WalletSynchronizer::*)(), WalletSynchronizer*> (
    __f=@0x555555f81710: (void (WalletSynchronizer::*)(WalletSynchronizer * const)) 0x5555556a0fd4 <WalletSynchronizer::mainLoop()>,
    __t=@0x555555f81708: 0x555577bf4930) at /usr/include/c++/8/bits/invoke.h:70
#18 0x00005555556a3ec2 in std::__invoke<void (WalletSynchronizer::*)(), WalletSynchronizer*> (
    __fn=@0x555555f81710: (void (WalletSynchronizer::*)(WalletSynchronizer * const)) 0x5555556a0fd4 <WalletSynchronizer::mainLoop()>)
    at /usr/include/c++/8/bits/invoke.h:89
#19 0x00005555556b2095 in std::thread::_Invoker<std::tuple<void (WalletSynchronizer::*)(), WalletSynchronizer*> >::_M_invoke<0ul, 1ul> (
    this=<optimized out>) at /usr/include/c++/8/thread:241
#20 0x00005555556b20a3 in std::thread::_Invoker<std::tuple<void (WalletSynchronizer::*)(), WalletSynchronizer*> >::operator() (this=<optimized out>)
    at /usr/include/c++/8/thread:250
#21 0x00005555556b20b3 in std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (WalletSynchronizer::*)(), WalletSynchronizer*> > >::_M_run (
    this=<optimized out>) at /usr/include/c++/8/thread:196
#22 0x00005555559a18bf in execute_native_thread_routine ()
#23 0x00007ffff79b76db in start_thread (arg=0x7ffff1a58700) at pthread_create.c:463
#24 0x00007ffff734271f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
```

**Comment by fentanyluser** [comment]
here is valgrind
```
==28158== Thread 4:
==28158== Invalid read of size 8
==28158==    at 0x25B287: std::vector<WalletTypes::RawTransaction, std::allocator<WalletTypes::RawTransaction> >::~vector() (stl_vector.h:434)
==28158==    by 0x25B2BA: WalletTypes::WalletBlockInfo::~WalletBlockInfo() (WalletTypes.h:218)
==28158==    by 0x25B2D2: std::_Head_base<0ul, WalletTypes::WalletBlockInfo, false>::~_Head_base() (tuple:120)
==28158==    by 0x25B830: std::_Tuple_impl<0ul, WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int>::~_Tuple_impl() (tuple:185)
==28158==    by 0x25B848: std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int>::~tuple() (tuple:556)
==28158==    by 0x25B98B: void __gnu_cxx::new_allocator<std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int> >::destroy<std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int> >(std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int>*) (new_allocator.h:140)
==28158==    by 0x25B996: void std::allocator_traits<std::allocator<std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int> > >::destroy<std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int> >(std::allocator<std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int> >&, std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int>*) (alloc_traits.h:487)
==28158==    by 0x25B9B1: std::vector<std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int>, std::allocator<std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int> > >::pop_back() (stl_vector.h:979)
==28158==    by 0x264090: std::priority_queue<std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int>, std::vector<std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int>, std::allocator<std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int> > >, OrderByArrivalIndex>::pop() (stl_queue.h:634)
==28158==    by 0x2640A6: ThreadSafePriorityQueue<std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int>, OrderByArrivalIndex>::pop_unsafe() (ThreadSafePriorityQueue.h:117)
==28158==    by 0x255650: WalletSynchronizer::mainLoop() (WalletSynchronizer.cpp:179)
==28158==    by 0x25869B: void std::__invoke_impl<void, void (WalletSynchronizer::*)(), WalletSynchronizer*>(std::__invoke_memfun_deref, void (WalletSynchronizer::*&&)(), WalletSynchronizer*&&) (invoke.h:73)
==28158==  Address 0x120d3630 is 0 bytes after a block of size 80 alloc'd
==28158==    at 0x4C3217F: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==28158==    by 0x4FDDF7: __gnu_cxx::new_allocator<std::__detail::_Hash_node<std::pair<Crypto::KeyImage const, Crypto::PublicKey>, true> >::allocate(unsigned long, void const*) (new_allocator.h:111)
==28158==    by 0x4FDE11: std::allocator_traits<std::allocator<std::__detail::_Hash_node<std::pair<Crypto::KeyImage const, Crypto::PublicKey>, true> > >::allocate(std::allocator<std::__detail::_Hash_node<std::pair<Crypto::KeyImage const, Crypto::PublicKey>, true> >&, unsigned long) (alloc_traits.h:436)
==28158==    by 0x500D24: std::__detail::_Hash_node<std::pair<Crypto::KeyImage const, Crypto::PublicKey>, true>* std::__detail::_Hashtable_alloc<std::allocator<std::__detail::_Hash_node<std::pair<Crypto::KeyImage const, Crypto::PublicKey>, true> > >::_M_allocate_node<std::piecewise_construct_t const&, std::tuple<Crypto::KeyImage const&>, std::tuple<> >(std::piecewise_construct_t const&, std::tuple<Crypto::KeyImage const&>&&, std::tuple<>&&) (hashtable_policy.h:2060)
==28158==    by 0x5010F5: std::__detail::_Map_base<Crypto::KeyImage, std::pair<Crypto::KeyImage const, Crypto::PublicKey>, std::allocator<std::pair<Crypto::KeyImage const, Crypto::PublicKey> >, std::__detail::_Select1st, std::equal_to<Crypto::KeyImage>, std::hash<Crypto::KeyImage>, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Prime_rehash_policy, std::__detail::_Hashtable_traits<true, false, true>, true>::operator[](Crypto::KeyImage const&) (hashtable_policy.h:725)
==28158==    by 0x501126: std::unordered_map<Crypto::KeyImage, Crypto::PublicKey, std::hash<Crypto::KeyImage>, std::equal_to<Crypto::KeyImage>, std::allocator<std::pair<Crypto::KeyImage const, Crypto::PublicKey> > >::operator[](Crypto::KeyImage const&) (unordered_map.h:973)
==28158==    by 0x4F3E70: SubWallets::storeTransactionInput(Crypto::PublicKey, WalletTypes::TransactionInput) (SubWallets.cpp:428)
==28158==    by 0x254F34: WalletSynchronizer::completeBlockProcessing(WalletTypes::WalletBlockInfo const&, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > > const&) (WalletSynchronizer.cpp:390)
==28158==    by 0x25556A: WalletSynchronizer::mainLoop() (WalletSynchronizer.cpp:178)
==28158==    by 0x25869B: void std::__invoke_impl<void, void (WalletSynchronizer::*)(), WalletSynchronizer*>(std::__invoke_memfun_deref, void (WalletSynchronizer::*&&)(), WalletSynchronizer*&&) (invoke.h:73)
==28158==    by 0x2586D0: std::__invoke_result<void (WalletSynchronizer::*)(), WalletSynchronizer*>::type std::__invoke<void (WalletSynchronizer::*)(), WalletSynchronizer*>(void (WalletSynchronizer::*&&)(), WalletSynchronizer*&&) (invoke.h:95)
==28158==    by 0x267166: decltype (__invoke((_S_declval<0ul>)(), (_S_declval<1ul>)())) std::thread::_Invoker<std::tuple<void (WalletSynchronizer::*)(), WalletSynchronizer*> >::_M_invoke<0ul, 1ul>(std::_Index_tuple<0ul, 1ul>) (thread:234)
==28158==
==28158== Invalid read of size 8
==28158==    at 0x25A959: std::vector<CryptoNote::KeyInput, std::allocator<CryptoNote::KeyInput> >::~vector() (stl_vector.h:434)
==28158==    by 0x25AA7B: WalletTypes::RawTransaction::~RawTransaction() (WalletTypes.h:129)
==28158==    by 0x25B22D: void std::_Destroy<WalletTypes::RawTransaction>(WalletTypes::RawTransaction*) (stl_construct.h:98)
==28158==    by 0x25B251: void std::_Destroy_aux<false>::__destroy<WalletTypes::RawTransaction*>(WalletTypes::RawTransaction*, WalletTypes::RawTransaction*) (stl_construct.h:108)
==28158==    by 0x25B268: void std::_Destroy<WalletTypes::RawTransaction*>(WalletTypes::RawTransaction*, WalletTypes::RawTransaction*) (stl_construct.h:137)
==28158==    by 0x25B273: void std::_Destroy<WalletTypes::RawTransaction*, WalletTypes::RawTransaction>(WalletTypes::RawTransaction*, WalletTypes::RawTransaction*, std::allocator<WalletTypes::RawTransaction>&) (stl_construct.h:206)
==28158==    by 0x25B295: std::vector<WalletTypes::RawTransaction, std::allocator<WalletTypes::RawTransaction> >::~vector() (stl_vector.h:434)
==28158==    by 0x25B2BA: WalletTypes::WalletBlockInfo::~WalletBlockInfo() (WalletTypes.h:218)
==28158==    by 0x25B2D2: std::_Head_base<0ul, WalletTypes::WalletBlockInfo, false>::~_Head_base() (tuple:120)
==28158==    by 0x25B830: std::_Tuple_impl<0ul, WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int>::~_Tuple_impl() (tuple:185)
==28158==    by 0x25B848: std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int>::~tuple() (tuple:556)
==28158==    by 0x25B98B: void __gnu_cxx::new_allocator<std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int> >::destroy<std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int> >(std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int>*) (new_allocator.h:140)
==28158==  Address 0xba8cd91b6ccb41bc is not stack'd, malloc'd or (recently) free'd
==28158==
==28158==
==28158== Process terminating with default action of signal 11 (SIGSEGV)
==28158==  General Protection Fault
==28158==    at 0x25A959: std::vector<CryptoNote::KeyInput, std::allocator<CryptoNote::KeyInput> >::~vector() (stl_vector.h:434)
==28158==    by 0x25AA7B: WalletTypes::RawTransaction::~RawTransaction() (WalletTypes.h:129)
==28158==    by 0x25B22D: void std::_Destroy<WalletTypes::RawTransaction>(WalletTypes::RawTransaction*) (stl_construct.h:98)
==28158==    by 0x25B251: void std::_Destroy_aux<false>::__destroy<WalletTypes::RawTransaction*>(WalletTypes::RawTransaction*, WalletTypes::RawTransaction*) (stl_construct.h:108)
==28158==    by 0x25B268: void std::_Destroy<WalletTypes::RawTransaction*>(WalletTypes::RawTransaction*, WalletTypes::RawTransaction*) (stl_construct.h:137)
==28158==    by 0x25B273: void std::_Destroy<WalletTypes::RawTransaction*, WalletTypes::RawTransaction>(WalletTypes::RawTransaction*, WalletTypes::RawTransaction*, std::allocator<WalletTypes::RawTransaction>&) (stl_construct.h:206)
==28158==    by 0x25B295: std::vector<WalletTypes::RawTransaction, std::allocator<WalletTypes::RawTransaction> >::~vector() (stl_vector.h:434)
==28158==    by 0x25B2BA: WalletTypes::WalletBlockInfo::~WalletBlockInfo() (WalletTypes.h:218)
==28158==    by 0x25B2D2: std::_Head_base<0ul, WalletTypes::WalletBlockInfo, false>::~_Head_base() (tuple:120)
==28158==    by 0x25B830: std::_Tuple_impl<0ul, WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int>::~_Tuple_impl() (tuple:185)
==28158==    by 0x25B848: std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int>::~tuple() (tuple:556)
==28158==    by 0x25B98B: void __gnu_cxx::new_allocator<std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int> >::destroy<std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int> >(std::tuple<WalletTypes::WalletBlockInfo, std::vector<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput>, std::allocator<std::tuple<Crypto::PublicKey, WalletTypes::TransactionInput> > >, unsigned int>*) (new_allocator.h:140)
==28158==

```

**Comment by brandonlehmann** [comment]
At this point I presume you’ve not only synced the wallet but also optimized it and churned the funds in it to avoid having an excessive amount of data in the wallet. 

Issue is being closed due to inactivity as it cannot be reproduced at this time.
